pkg/controller/utils/utils.go

```diff
@@ -71,3 +75,22 @@ func AddFinalizer(object metav1.Object, finalizer string) {
 	finalizers.Insert(finalizer)
 	object.SetFinalizers(finalizers.List())
 }
+
+// LogAwsError formats and logs aws error and returns if err was an awserr
+func LogAwsError(logger logr.Logger, errMsg string, returnError error, err error) bool {
```

Comment:
Does this need to return a bool?

Reply:
For use cases like [this](https://github.com/openshift/aws-account-operator/pull/194/files#diff-26f5cba4465c7a85a44a0dcf6ac15793R183) it's helpful.

Reply:
But this logic is already handled in this `LogAwsError` function. The example you gave was written because not error checking is really done, but in line 81, you do this exact thing.

Reply:
If this function doesn't return a bool. Then this specific case can't be covered.

Because the other solutions here are to change [this one call](https://github.com/openshift/aws-account-operator/pull/194/files#diff-26f5cba4465c7a85a44a0dcf6ac15793R183) to not handle the case where `err.(awserr.Error)` fails. Or call the function the clause after it passes. Which is a redundant check.

Reply:
All you would need to do for that case is fix this:
`if utils.LogAwsError(log, "", err, err) != true {`
to
`utils.LogAwsError(log, "Error listing managed AWS policies", err, err)`

Reply:
Ack